pennylane/templates/subroutines/amplitude_amplification.py

```diff
@@ -169,6 +169,8 @@ def compute_decomposition(**kwargs):
         else:
             for _ in range(iters):
                 ops.append(O)
+                if qml.QueuingManager.recording():
+                    qml.apply(O)
```

Comment:
Currently, decompositions need to return the correct list but also need to queue the operations of the decomposition. `ops.append(O)` only does the former.

Reply:
Good catch!